frontend/src/metabase/embedding/hooks/use-sharing-modal.ts

```diff
@@ -0,0 +1,57 @@
+import { useCallback, useState } from "react";
+
+import type {
+  DashboardSharingModalType,
+  QuestionSharingModalType,
+} from "metabase/embedding/components/SharingMenu/types";
+import { GUEST_EMBED_EMBEDDING_TYPE } from "metabase/embedding/constants";
+import { useOpenEmbedJsWizard } from "metabase/embedding/hooks/use-open-embed-js-wizard";
+import { useDispatch } from "metabase/lib/redux";
+import type {
+  EmbedResource,
+  EmbedResourceType,
+} from "metabase/public/lib/types";
+import { setOpenModal } from "metabase/redux/ui";
+
+export const useSharingModal = <
+  TModalType extends DashboardSharingModalType | QuestionSharingModalType,
```

Comment:
(I didn't expected the public links to also be managed by this `useSharingModal` hook since I didn't consider the public links as modals.)